sdk/storage/inc/datalake/path_client.hpp

```diff
@@ -0,0 +1,249 @@
+     *                 the order in which it is appended to the file.
+     *                 The value must be the offset where the data is to be appended.
+     *                 Uploaded data is not immediately flushed, or written, to the file. To flush,
+     *                 the previously uploaded data must be contiguous, the offset parameter must
+     *                 be specified and equal to the length of the file after all data has been
+     *                 written, and there must not be a request entity body included with the
+     *                 request.
+     * @param options Optional parameters to flush data to the resource the path points to.
+     * @return PathFlushDataResponse
+     */
+    PathFlushDataResponse FlushData(
+        int64_t offset,
+        const PathFlushDataOptions& options = PathFlushDataOptions()) const;
+
+    /**
+     * @brief Sets the owner, group, permissions, or access control list for a file or directory.
+     *        Note that Hierarchical Namespace must be enabled for the account in order to use
+     *        access control. Also note that the Access Control List (ACL) includes permissions for
+     *        the owner, owning group, and others, so the x-ms-permissions and x-ms-acl request
+     *        headers are mutually exclusive.
+     * @param options Optional parameters to set an access control to the resource the path points
+     *                to.
+     * @return PathFlushDataResponse
+     */
+    PathSetAccessControlResponse SetAccessControl(
+        const SetAccessControlOptions& options = SetAccessControlOptions()) const;
+
+    /**
+     * @brief Sets POSIX access control rights on files and directories under given path
+     *        recursively.
+     * @param mode Mode PathSetAccessControlRecursiveMode::Set sets POSIX access control rights on
+     *             files and directories, PathSetAccessControlRecursiveMode::Modify modifies one or
+     *             more POSIX access control rights  that pre-exist on files and directories,
+     *             PathSetAccessControlRecursiveMode::Remove removes one or more POSIX access
+     *             control rights that were present earlier on files and directories
+     * @param options Optional parameters to set an access control recursively to the resource the
+     *                path points to.
+     * @return PathSetAccessControlRecursiveResponse
+     */
+    PathSetAccessControlRecursiveResponse SetAccessControlRecursive(
+        PathSetAccessControlRecursiveMode mode,
+        const SetAccessControlRecursiveOptions& options = SetAccessControlRecursiveOptions()) const;
+
+    /**
+     * @brief Sets the properties of a resource the path points to.
+     * @param options Optional parameters to set the properties to the resource the path points to.
+     * @return PathUpdateResponse
+     */
+    PathUpdateResponse SetProperties(
```

Comment:
Still feel strange that in Blob service this function is called SetHttpHeaders and in DataLake service this is called SetProperties, while they actually do the same thing. Please escalate this.

Reply:
Will bring this up.